apps/ewallet/lib/ewallet/web/paginator.ex

```diff
@@ -43,39 +44,120 @@ defmodule EWallet.Web.Paginator do
   """
   @spec paginate_attrs(Ecto.Query.t() | Ecto.Queryable.t(), map(), Ecto.Repo.t()) ::
           %__MODULE__{} | {:error, :invalid_parameter, String.t()}
-  def paginate_attrs(queryable, attrs, repo \\ Repo)
+  def paginate_attrs(queryable, attrs, allowed_page_record_fields \\ [], repo \\ Repo)
 
-  def paginate_attrs(queryable, %{"page" => page} = attrs, repo) when not is_integer(page) do
-    parse_string_param(queryable, attrs, "page", page, repo)
+  def paginate_attrs(queryable, %{"page" => page} = attrs, allowed_page_record_fields, repo)
+      when not is_integer(page) do
+    parse_string_param(queryable, attrs, "page", page, allowed_page_record_fields, repo)
   end
 
-  def paginate_attrs(_, %{"page" => page}, _repo) when is_integer(page) and page < 0 do
-    {:error, :invalid_parameter, "`page` must be non-negative integer"}
+  def paginate_attrs(
+        queryable,
+        %{"per_page" => per_page} = attrs,
+        allowed_page_record_fields,
+        repo
+      )
+      when not is_integer(per_page) do
+    parse_string_param(queryable, attrs, "per_page", per_page, allowed_page_record_fields, repo)
   end
 
-  def paginate_attrs(queryable, %{"per_page" => per_page} = attrs, repo)
-      when not is_integer(per_page) do
-    parse_string_param(queryable, attrs, "per_page", per_page, repo)
+  def paginate_attrs(_, %{"page" => _, "page_record_value" => _}, _, _) do
+    {:error, :invalid_parameter, "`page` cannot be used with `page_record_value`"}
+  end
+
```

Comment:
Should we also prevent "sort_by" and "page_record_value" be used together due to its ambiguity? For example:

With these conditions:

```
{
  "page_record_field": "address",
  "page_record_value": "addr2222"
  "sort_by": "created_at"
  "sort_dir": "asc"
}
```

And the data:

address | inserted_at
-------- | -----------
addr1111 | Jan 21
addr3333 | Jan 22
addr2222 | Jan 23

If it honors `address ASC`, it would return:

address | inserted_at
-------- | -----------
addr3333 | Jan 22
addr2222 | Jan 23

If it honors `created_at ASC`, it would return:

address | inserted_at
-------- | -----------
addr2222 | Jan 23

Reply:
I think we should allow to be used together. I'm thinking of it like this.
- `page_record_value` means `from where will we fetch`.
- `sort_by` means `how do we sort the fetched data`.

This is how sorting work when we do specify `sort_by` and don't:

|     sort_by     |              records sort by              |
|:---------------:|:--------------------------------:|
|    `create_at`    |     `created_at`  |
| _not specified_ |  `page_record_field`|

Reply:
Your table helps a lot! I think we should add a new document into https://github.com/omisego/ewallet/tree/master/docs/guides to explain the common request attributes we allow.

Reply:
Noted